autotest/ogr/ogr_gpkg.py

```diff
@@ -7988,71 +8018,65 @@ def test_ogr_gpkg_arrow_stream_numpy_detailed_spatial_filter(tmp_vsimem, layer_t
 
 
 ###############################################################################
 # Test opening a file in WAL mode on a read-only storage
 
 
 @pytest.mark.skipif(sys.platform != "linux", reason="Incorrect platform")
-def test_ogr_gpkg_immutable():
+def test_ogr_gpkg_immutable(tmp_path):
 
     if os.getuid() == 0:
         pytest.skip("running as root... skipping")
 
-    try:
-        os.mkdir("tmp/read_only_test_ogr_gpkg_immutable", 0o755)
+    os.mkdir(tmp_path / "read_only_test_ogr_gpkg_immutable", 0o755)
 
-        ds = ogr.GetDriverByName("GPKG").CreateDataSource(
-            "tmp/read_only_test_ogr_gpkg_immutable/test.gpkg"
-        )
-        ds.CreateLayer("foo")
-        ds.ExecuteSQL("PRAGMA journal_mode = WAL")
-        ds = None
+    ds = ogr.GetDriverByName("GPKG").CreateDataSource(
+        tmp_path / "read_only_test_ogr_gpkg_immutable/test.gpkg"
+    )
+    ds.CreateLayer("foo")
+    ds.ExecuteSQL("PRAGMA journal_mode = WAL")
+    ds = None
 
-        # Turn directory in read-only mode
-        os.chmod("tmp/read_only_test_ogr_gpkg_immutable", 0o555)
+    # Turn directory in read-only mode
+    os.chmod(tmp_path / "read_only_test_ogr_gpkg_immutable", 0o555)
 
-        with gdal.quiet_errors():
-            assert (
-                gdal.OpenEx(
-                    "tmp/read_only_test_ogr_gpkg_immutable/test.gpkg",
-                    gdal.OF_VECTOR | gdal.OF_UPDATE,
-                )
-                is None
-            )
-            assert (
-                gdal.OpenEx(
-                    "tmp/read_only_test_ogr_gpkg_immutable/test.gpkg",
-                    gdal.OF_VECTOR,
-                    open_options=["IMMUTABLE=NO"],
-                )
-                is None
+    with gdal.quiet_errors():
+        assert (
+            gdal.OpenEx(
+                tmp_path / "read_only_test_ogr_gpkg_immutable/test.gpkg",
+                gdal.OF_VECTOR | gdal.OF_UPDATE,
             )
-
-        gdal.ErrorReset()
+            is None
+        )
         assert (
             gdal.OpenEx(
-                "tmp/read_only_test_ogr_gpkg_immutable/test.gpkg",
+                tmp_path / "read_only_test_ogr_gpkg_immutable/test.gpkg",
                 gdal.OF_VECTOR,
-                open_options=["IMMUTABLE=YES"],
+                open_options=["IMMUTABLE=NO"],
             )
-            is not None
+            is None
         )
-        assert gdal.GetLastErrorMsg() == ""
 
-        gdal.ErrorReset()
-        with gdal.quiet_errors():
-            assert (
-                ogr.Open("tmp/read_only_test_ogr_gpkg_immutable/test.gpkg") is not None
-            )
-        assert gdal.GetLastErrorMsg() != ""
+    gdal.ErrorReset()
+    assert (
+        gdal.OpenEx(
+            tmp_path / "read_only_test_ogr_gpkg_immutable/test.gpkg",
+            gdal.OF_VECTOR,
+            open_options=["IMMUTABLE=YES"],
+        )
+        is not None
+    )
+    assert gdal.GetLastErrorMsg() == ""
 
-    finally:
```

Comment:
does pytest manage to remove the temporary directory despite our making it read-only ?

Reply:
Great question. Apparently yes, it does:
```
ls /tmp/pytest-of-dan/pytest-86/test_ogr_gpkg_immutable0/read_only_test_ogr_gpkg_immutable
# test.gpkg

pytest ogr/ogr_gpkg.py -k immutable
ls /tmp/pytest-of-dan/pytest-86/test_ogr_gpkg_immutable0/read_only_test_ogr_gpkg_immutable
# ls: cannot access '/tmp/pytest-of-dan/pytest-86/test_ogr_gpkg_immutable0/read_only_test_ogr_gpkg_immutable': No such file or directory
```